keras/engine/topology.py

```diff
@@ -2485,6 +2487,20 @@ def load_weights_from_hdf5_group(self, f):
                 weight_value_tuples += zip(symbolic_weights, weight_values)
             K.batch_set_value(weight_value_tuples)
 
+        backend = f.attrs.get('keras_backend')
+        if backend and backend != K._BACKEND:
+            # Weights were originally serialized using a different backend. TensorFlow and Theano
+            # use different implementations for convolutions, which need to be converted.
+            weight_value_tuples = []
+            for layer in flattened_layers:
```

Comment:
This loop should be integrated with the first one. Otherwise we need two calls to `batch_set_values`, which is less efficient. Weights should be converted in Numpy-space before being set.